superset-frontend/src/explore/controls.jsx

```diff
@@ -204,9 +204,9 @@ export const controls = {
     label: t('Datasource'),
     default: null,
     description: null,
-    mapStateToProps: (state, control, actions) => ({
-      datasource: state.datasource,
-      onDatasourceSave: actions ? actions.setDatasource : () => {},
+    mapStateToProps: ({ datasource }) => ({
+      datasource,
+      isEditable: !!datasource,
```

Comment:
Idea for future plan: this `isEditable` attribute should probably be a datasource attribute, based on user access or datasource metadata.

Reply:
Could you please set this isEditable a datasource property? in airbnb we do have use-case that don't allow user to edit certain datasources.

Reply:
Let's do that in a future refactor.